app/services/booking_responder.rb

```diff
@@ -0,0 +1,26 @@
+class BookingResponder
+  def initialize(booking_response)
+    @booking_response = booking_response
+  end
+
+  def respond!
+    if booking_response.slot_selected?
+      accept!
+    else
+      reject!
+    end
+  end
+
+private
+
+  attr_reader :booking_response
+  delegate :visit, to: :booking_response
```

Comment:
Although this delegation is declared beneath `private` it will actually expose the `visit` instance publicly.